### ISSUE TITLE
match/exclude in interceptors don't correctly read the controllername when having hyphenated url converter

### ISSUE THREAD

**Issue by olavgg** [grails/grails-core#11165]
### Steps to Reproduce

1. open LogEnvController index action

### Expected Behaviour
In Grails when you refer to controller names, you use non-hyphenated references. For example with the createLink taglib or in URLMappings. 

### Actual Behaviour
But Interceptors match only work if you match with the hyphenated uri reference of the controller name.
When I see in the code (org.grails.plugins.web.interceptors.UrlMappingMatcher) it only seem to run the match against the request uri.

So in conclusion:
match(controller: "logEnvSwitch", action: "*") does not work
match(controller: "log-env-switch", action: "*") work

Consistent behavior would be support for both?

### Environment Information

- **Operating System**: Ubuntu 18.04 64bits
- **Grails Version:** 3.3.8
- **JDK Version:** OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-1ubuntu0.18.04.1-b13)

### Example Application

https://github.com/olavgg/hyphenatedinterceptorbug

**Comment by olavgg** [comment]
I agree with you, I don't think hyphenated references would work for taglibs or in URL mapping, which means supporting it in interceptors will only add more confusion.

**Comment by olavgg** [comment]
I want to give an update as I am working on resolving this.

controllerName, actionName and namespace also prints the hyphenated names inside the interceptor. This may be the reason the matching is also not working, though I have not managed to figure out that part yet.

Anyway in DefaultUrlMappingInfo in the getControllerName, getActionName and getNamspace methods, I have added a hyphenated to camel case converter https://github.com/olavgg/grails-core/commit/6989cf3412c66be001a2e68cac66e7a1714ce930#diff-4d4ad4c96b2a8d7368334cdf04d21fdb

This seems to resolve the issue with controllerName, actionName and controllerNamespace inside the interceptor.

I feel I am in deep water here with my changes, I know there is some code duplication here but would like some feedback if this looks correct before refactoring it.

The bug with matching is something I need to look further into.
